pos_loyalty_exclude/models/loyalty_rule.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.osv import expression
+
+
+class LoyaltyRule(models.Model):
+    _inherit = "loyalty.rule"
+
+    @api.depends("product_ids", "product_category_id", "product_tag_id")
+    def _compute_valid_product_ids(self):
+        super()._compute_valid_product_ids()
+        excluded_products = self.env["product.product"].search_count(
+            [("loyalty_exclude", "=", True), ("available_in_pos", "=", True)]
+        )
+        # this check ensures that it only runs when there are loyalty-excluded products
+        if excluded_products:
+            for rule in self:
+                # exclude product when no config in loyalty rule
+                if not (
+                    rule.product_ids
+                    or rule.product_category_id
+                    or rule.product_tag_id
+                    or rule.product_domain not in ("[]", "[['sale_ok', '=', True]]")
```

Comment:
is [['sale_ok', '=', True]] intended or do you meant [('sale_ok', '=', True)]

Reply:
Hello @Ricardoalso, it is intended as it is in https://github.com/odoo/odoo/blob/def7c93c6c05e1814d6221cf90e93bdd8867e473/addons/pos_loyalty/models/loyalty_rule.py#L29